x-pack/solutions/security/plugins/security_solution/common/api/endpoint/actions/response_actions/kill_process/kill_process.schema.yaml

```
@@ -35,11 +41,49 @@ components:
           properties:
             parameters:
               oneOf:
-                - $ref: "../../../model/schema/common.schema.yaml#/components/schemas/Pid"
-                - $ref: "../../../model/schema/common.schema.yaml#/components/schemas/EntityId"
+                - type: object
+                  properties:
+                    pid:
+                      type: integer
+                      description: "The process ID (PID) of the process to terminate."
+                      example: 123
+                      minimum: 1
+                - type: object
+                  properties:
+                    entity_id:
+                      type: string
+                      description: "The entity ID of the process to terminate."
+                      example: "abc123"
+                      minLength: 1
                 - type: object
                   properties:
                     process_name:
                       type: string
+                      description: Valid for SentinelOne agent type only.
```

```suggestion
                      description: "The name of the process to terminate. Valid for SentinelOne agent type only."
```

I'm also thinking if we should add some text to the description of the other two parameters above to indicate they are only valid with an agent type of `endpoint`